ovirt_imageio/_internal/qemu_nbd.py

```diff
@@ -245,6 +253,11 @@ def _preexec_fn(self):
         if self.block_signals:
             signal.pthread_sigmask(signal.SIG_BLOCK, self.block_signals)
 
+    def _is_block_dev(self):
+        """
+        Checks if a path is of a block device
+        """
+        return stat.S_ISBLK(os.lstat(self.image).st_mode)
```

Comment:
os.lstat() does not follow symbolic links, which is the worng thing to do since /dev/xxx may be symlink to the real device (e.g. /dev/dm-42). os.stat() is the right function.